sor-api/src/main/java/com/bazaarvoice/emodb/sor/api/CompactionControlSource.java

```diff
@@ -0,0 +1,20 @@
+package com.bazaarvoice.emodb.sor.api;
+
+import com.bazaarvoice.emodb.auth.proxy.Credential;
+
+import java.util.List;
+import java.util.Map;
+
+public interface CompactionControlSource {
+
+    void updateStashTime(String id, long timestamp, List<String> placements, long expiredTimestamp, Boolean remote);
+
+    void deleteStashTime(String id);
+
+    StashRunTimeInfo getStashTime(String id);
+
+    Map<String, StashRunTimeInfo> getStashTimes();
+
+    long getOldStashTime();
```

Comment:
Open question: This seems like more of a convenience method than providing any core functionality.  Would it make more sense to remove this and have the callers use `getStashTimes()` directly and manipulate however they need?